### ISSUE TITLE
Ampersand is not escaped in "Open file in another instance" command

### ISSUE THREAD

**Issue by galvanopus** [notepad-plus-plus/notepad-plus-plus#5638]
### Description of the Issue
Notepad++ 7.6.6 clean install.
Ampersand is not escaped in "Open file in another instance" command:
`<Command name="Open file in another instance" Ctrl="no" Alt="yes" Shift="no" Key="117">$(NPP_FULL_FILE_PATH) $(CURRENT_WORD) -nosession -multiInst</Command>`

### Steps to Reproduce the Issue
1. Create file `C:\Users\galvanopus\Documents\Research & Development.md`.
2. Open Notepad++, paste full file name to a new document, select it and press Alt + F6.

### Expected Behavior
File `C:\Users\galvanopus\Documents\Research & Development.md` should be opened in a new instance.

### Actual Behavior
3 error messages are shown:
`"C:\Users\galvanopus\Documents\Research" doesn't exist. Create it?`
`"C:\Program Files\Notepad++\&" doesn't exist. Create it?`
`"C:\Program Files\Notepad++\Development.md" doesn't exist. Create it?`

### Possible Fix
Surround `$(CURRENT_WORD)` with `&quot;`:
`<Command name="Open file in another instance" Ctrl="no" Alt="yes" Shift="no" Key="117">$(NPP_FULL_FILE_PATH) &quot;$(CURRENT_WORD)&quot; -nosession -multiInst</Command>`

The suggested fix has the issue with quoted selection. The same 3 error messages appear if selecting `"C:\Users\galvanopus\Documents\Research & Development.md"` (do you see the quotes?) and pressing Alt + F6. It might worth introducing `Quote_CURRENT_WORD` command parameter. If its value is `Yes` then all left and right quotes are removed from CURRENT_WORD (there can be more than one quote pair from each side) and exactly one pair is added from left and right sides. Here are the examples:

| Before  | After |
| --- | --- |
| `C:\Research & Development.md` | `"C:\Research & Development.md"` |
| `"C:\Research & Development.md"` | `"C:\Research & Development.md"` |
| `""C:\Research & Development.md"` | `"C:\Research & Development.md"` |
| `"""C:\Research & Development.md"` | `"C:\Research & Development.md"` |
| `"""C:\Research & Development.md""` | `"C:\Research & Development.md"` |
| `"""C:\Research & Development.md"""` | `"C:\Research & Development.md"` |
| `"""C:\Research & Development.md""""` | `"C:\Research & Development.md"` |

**Comment by wqweto** [comment]
https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/

tl;dr

You cannot randomly slap quotes around  command line arguments and prey that `CommandLineToArgvW` API will parse it correctly in target application.

There is an `ArgvQuote` function included in the blog post with *correct* escape implementation in C/C++.
